iot_config/src/gateway/tracker.rs

```
@@ -0,0 +1,79 @@
+use crate::gateway::{db::Gateway, metadata_db::IOTHotspotInfo};
+use futures::stream::TryChunksError;
+use futures_util::TryStreamExt;
+use sqlx::{Pool, Postgres};
+use std::time::{Duration, Instant};
+use task_manager::ManagedTask;
+
+const EXECUTE_DURATION_METRIC: &str =
+    concat!(env!("CARGO_PKG_NAME"), "-", "tracker-execute-duration");
+
+pub struct Tracker {
+    pool: Pool<Postgres>,
+    metadata: Pool<Postgres>,
+    interval: Duration,
+}
+
+impl ManagedTask for Tracker {
+    fn start_task(
+        self: Box<Self>,
+        shutdown: triggered::Listener,
+    ) -> task_manager::TaskLocalBoxFuture {
+        task_manager::spawn(self.run(shutdown))
+    }
+}
+
+impl Tracker {
+    pub fn new(pool: Pool<Postgres>, metadata: Pool<Postgres>, interval: Duration) -> Self {
+        Self {
+            pool,
+            metadata,
+            interval,
+        }
+    }
+
+    async fn run(self, mut shutdown: triggered::Listener) -> anyhow::Result<()> {
+        tracing::info!("starting with interval: {:?}", self.interval);
+        let mut interval = tokio::time::interval(self.interval);
+
+        loop {
+            tokio::select! {
+                biased;
+                _ = &mut shutdown => break,
+                _ = interval.tick() => {
+                    if let Err(err) = execute(&self.pool, &self.metadata).await {
```

We don't want to propagate this err and crash the service if it happens?